src/click-outside.directive.ts

```diff
@@ -52,12 +54,15 @@ export class ClickOutsideDirective implements OnInit, OnChanges, OnDestroy {
     }
 
     this._events.forEach(e => document.body.removeEventListener(e, this._onClickBody));
+    if (this.emitOnBlur) {
```

Comment:
We should keep a flag to see if it's been attached or not, in case the listener is added, but then the option gets turned off for some reason.

Reply:
Hmm, if I understand you correctly, then there should be some general logic to handle this case, e.g. for `attachOutsideOnClick` and `clickOutsideEvents`, right? Besides, firing `window.removeEventListener` when the listener hasn't been added is safe.

Reply:
My concern is that if the event listener is added but then the directive is destroyed but the listener is still there. It should almost never happen, but I'd rather it not be emitting phantom events in that case.